<launch>
  <!--
      online segmentation using .npy & SqueezeSeg model
      this script can
                    1. read all .npy file from lidar_2d folder
                    2. predict label from SqueezeSeg model using tensorflow
                    3. publish to 'sqeeuze_seg/points' topic
  -->
  <param name="npy_path" type="string" value="/media/cyber-z/E/npy/" />
  <param name="npy_file_list" type="string" value="$(find squeezeseg_ros)/script/data/ImageSet/test.txt" />
  <param name="pub_topic" type="string" value="/squeeze_seg/points" />
  <param name="sub_topic" type="string" value="/kitti/velo/pointcloud" />
  <!-- should be the path for checkpoint -->
  <param name="input_type" type="string" value="bag" />
  <param name="checkpoint" type="string" value="$(find squeezeseg_ros)/script/data/SqueezeSeg/model.ckpt-23000" />
  <param name="gpu" type="string" value="0" />
  <param name="x_channel" type="string" value="x" />
  <param name="y_channel" type="string" value="y" />
  <param name="z_channel" type="string" value="z" />
  <param name="i_channel" type="string" value="i" />

  <node pkg="squeezeseg_ros" type="squeezeseg_ros_node.py" name="squeezeseg_ros_node" output="screen" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find squeezeseg_ros)/rviz/squeezeseg_ros.rviz" />

</launch>